assets/js/modules/tagmanager/setup.js

```diff
@@ -189,51 +225,61 @@ class TagmanagerSetup extends Component {
 				usageContext,
 			};
 
-			let errorCode = false;
-			let errorMsg = '';
 			const { accounts, containers } = await data.get( TYPE_MODULES, 'tagmanager', 'accounts-containers', queryArgs );
 
-			if ( ! selectedAccount && 0 === accounts.length ) {
-				errorCode = 'accountEmpty';
-				errorMsg = __(
-					'We didn’t find an associated Google Tag Manager account, would you like to set it up now? If you’ve just set up an account please re-fetch your account to sync it with Site Kit.',
-					'google-site-kit'
-				);
-			}
-
-			// Verify if user has access to the selected account.
-			if ( selectedAccount && ! accounts.find( ( account ) => account.accountId === selectedAccount ) ) { // Capitalization rule exception: `accountId` is a property of an API returned value.
-				data.invalidateCacheGroup( TYPE_MODULES, 'tagmanager', 'accounts-containers' );
-				errorCode = 'insufficientPermissions';
-				errorMsg = __( 'You currently don\'t have access to this Google Tag Manager account. You can either request access from your team, or remove this Google Tag Manager snippet and connect to a different account.', 'google-site-kit' );
-			}
+			this.validateAccounts( accounts, selectedAccount );
 
 			// If the selectedContainer is not in the list of containers, clear it.
-			if ( selectedContainer && ! containers.find( ( container ) => container.publicId === selectedContainer ) ) {
+			if ( isValidContainerID( selectedContainer ) && ! containers.find( ( container ) => container.publicId === selectedContainer ) ) { /* Capitalization rule exception: `publicId` is a property of an API returned value. */
 				selectedContainer = null;
 			}
 
-			if ( this._isMounted ) {
-				this.setState( {
-					isLoading: false,
-					accounts,
-					selectedAccount: selectedAccount || get( containers, [ 0, 'accountId' ] ), // Capitalization rule exception: `accountId` is a property of an API returned value.
-					containers,
-					selectedContainer: selectedContainer || get( containers, [ 0, 'publicId' ] ), // Capitalization rule exception: `publicId` is a property of an API returned value.
-					refetch: false,
-					errorCode,
-					errorMsg,
-				} );
-			}
+			this.setState( {
+				isLoading: false,
+				accounts,
+				selectedAccount: selectedAccount || get( containers, [ 0, 'accountId' ] ), // Capitalization rule exception: `accountId` is a property of an API returned value.
+				containers,
+				selectedContainer: selectedContainer || get( containers, [ 0, 'publicId' ] ), // Capitalization rule exception: `publicId` is a property of an API returned value.
```

Comment:
This shouldn't be addressed by this PR, but I think it's a bit strange that we just select the first account/container. I think we should revisit this on a broader scope as part of #1101 and following issues. Maybe only pre-select if
* either there is specific logic that determined this selection as likely (e.g. URL matching)
* or there is just one list entry (no point in selecting)